creator-node/src/services/sync/secondarySyncFromPrimary.js

```diff
@@ -27,44 +29,40 @@ const handleSyncFromPrimary = async (
   const start = Date.now()
   let returnValue = {}
 
-  logger.info('begin nodesync for ', walletPublicKeys, 'time', start)
+  genericLogger.info('begin nodesync for ', wallet, 'time', start)
 
-  /**
-   * Ensure access to each wallet, then acquire redis lock for duration of sync
-   * @notice - there's a bug where logPrefix is set to the last element of `walletPublicKeys` - this code only works when `walletPublicKeys.length === 1` 🤦‍♂️
-   */
-  let logPrefix
-  for (const wallet of walletPublicKeys) {
-    logPrefix = `[secondarySyncFromPrimary][wallet=${wallet}]`
-    try {
-      await redis.WalletWriteLock.acquire(
-        wallet,
-        redis.WalletWriteLock.VALID_ACQUIRERS.SecondarySyncFromPrimary
-      )
-    } catch (e) {
-      return {
-        error: new Error(
-          `Cannot change state of wallet ${wallet}. Node sync currently in progress.`
-        ),
-        result: 'failure_sync_in_progress'
-      }
+  const logPrefix = `[secondarySyncFromPrimary][wallet=${wallet}]`
+  try {
+    await redis.WalletWriteLock.acquire(
+      wallet,
+      redis.WalletWriteLock.VALID_ACQUIRERS.SecondarySyncFromPrimary
+    )
+  } catch (e) {
+    return {
+      error: new Error(
+        `Cannot change state of wallet ${wallet}. Node sync currently in progress.`
+      ),
+      result: 'failure_sync_in_progress'
     }
   }
 
   /**
    * Perform all sync operations, catch and log error if thrown, and always release redis locks after.
    */
-  try {
-    const wallet = walletPublicKeys[0]
+  const forceResync = await shouldForceResync(
+    { libs, logContext },
+    forceResyncConfig
+  )
```

Comment:
just had a thought - would be great to record a metric here (similar to rest of function) if we reject a sync bc of some error here
would require a bit of work since we want to differentiate btwn an error vs just forceResync flag not being passed?

Reply:
i dont know if tracking the error state is worthwhile because we should never not be passing forceResync because our usage for this config should be always correct. i think we should instead be tracking the number of times forceResync=true/false

Reply:
agreed, tracking forceResync true/false is probably valuable
but also - point of metrics is to see when things happen unexpectedly, and if there are a ton of signature failures we'd want to track that

Reply:
this can be a followup task